### PR TITLE
test(plugin): test for fix of otel reconfigure issue #KAG-1061

### PR DESCRIPTION
**Pull request by StarlightIbuki**
### Summary

[Improvement of batch queue](https://github.com/Kong/kong/issues/10172) fixed the OTEL reconfigure issue.
This PR adds a test to verify the fix.

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] <s>There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE</s> N/A

### Full changelog

* Cherry-pick coroutine-based HTTP mock helper function
* Regression test for the issue

### Issue reference

Test of KAG-1061

### CODE REVIEW

**Review by hanshuebner — CHANGES_REQUESTED** [review]
Please update the title of the PR to describe the issue.  Also update the summary in the description to include the problem, the root cause and the solution.  It currently seems to say that this PR fixes an issue where in reality the issue was fixed by the queueing changes and what it contains is just the regression test.

File: spec/helpers.lua [hunk 1]
```diff
@@ -1345,6 +1345,183 @@ local function http_server(port, opts)
 end
```
[the remark this answers is not in the document]
Comment (StarlightIbuki) [reply spec/helpers.lua:1347]: This is not a duplication. `http_server` uses a thread and `http_mock` uses a coroutine. The coroutine implementation is more deterministic and thus less flaky, but we choose to not retire `http_server` for now as it involves a lot of changes in the test suit (plus I need to convince people to switch to this).

[the remark this answers is not in the document]
Comment (hanshuebner) [reply spec/helpers.lua:1347]: Please add a comment to the threads based implementation, pointing out that `http_mock` should be used instead, and open a KAG ticket "Replace usage of `helpers.http_server` by `helpers.http_mock`".

[the remark this answers is not in the document]
Comment (StarlightIbuki) [reply spec/helpers.lua:1347]: Please refer to https://konghq.atlassian.net/browse/KAG-1148

[the remark this answers is not in the document]
Comment (hanshuebner) [reply spec/helpers.lua:1349]: I would still like to see a deprecation comment in helpers.lua to make people aware of the issue.

[the remark this answers is not in the document]
Comment (StarlightIbuki) [reply spec/helpers.lua:1349]: @hanshuebner I've added comments to `http_server`.

**Review by hanshuebner — APPROVED** [review]

**Pull request merged**